prometheus/go_collector_latest_go120_test.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !go1.21
+// +build !go1.21
+
+package prometheus
+
+var expMetrics = map[string]string{
```

Comment:
Nice! Unfortunately this filename does little sense (both latest and 120 in name). We can rename, but I wonder if there's a way to put this into single file with other metrics. 

I think @ArthurSens is right, we might want to check how we could integrate with https://github.com/prometheus/client_golang/blob/main/prometheus/collectors/gen_go_collector_set.go perhaps, as it was created for same reason. Not sure if it's integrated in our CI, worth to check...

Reply:
Ok, now I understand. Yes, ideally that map should be included in the corresponding Go version autogenerated files.